### ISSUE TITLE
.shp always appended to filename, even if it already has .shp extension (gridgen module)

### ISSUE THREAD

**Issue by OnnoEbbens** [modflowpy/flopy#2024]
I use a shapefile as a refinement feature in Gridgen. When I specify a filename with the '.shp' extension, let's say `shapefile='my_shape.shp'`, I get an error, `Shapefile does not exist: my_shape.shp.shp`. The is because the .shp extension is always added to the filename in de gridgen.py module line 446:

```
sn = os.path.join(self.model_ws, f"{shapefile}.shp")
assert os.path.isfile(sn), f"Shapefile does not exist: {sn}"
``` 

As a quick solution I propose to only add the 'shp' extension if it does not have one already. The code can be something like this:
```
if not shapefile.endswith('.shp'):
    shapefile += '.shp'
sn = os.path.join(self.model_ws, shapefile)
assert os.path.isfile(sn), f"Shapefile does not exist: {sn}"
```

Another thing I noticed is that the shapefile is assumed to be in the model workspace or specified relative to the model workspace. It would be nice to allow for an absolute path of the shapefile.

I can draft a PR for this if you guys like the proposed solution.

This issue was originally posted here: https://github.com/gwmod/nlmod/issues/303

**Comment by wpbonelli** [comment]
@OnnoEbbens this was raised/partially patched already #2021 but I think the solution you describe is ideal. I had already started to work on it, but thanks much for the PR offer, would you mind having a look at #2026?

**Comment by OnnoEbbens** [comment]
Thanks for fixing this. I like your PR, it solves all the things requested.
